lib/utils/utils.js

```diff
@@ -64,8 +66,49 @@ function setConsoleLogPrefix(prefix) {
     };
 }
 
+// This should be roughly similar to ARG_MAX, some systems is 131072 bytes
+// but we're going to be very conservative and cap the total size much lower than that to be safe.
+// POSIX suggestion is to go 2048 lower than the ARG_MAX value, as per https://www.in-ulm.de/~mascheck/various/argmax/
+const CMD_SIZE_LIMIT = 128 * 1024;
+const VAR_PATH = "./vars";
+
+/**
+ * Only allows a specific number of bytes for variable names and values.  Any variable exceeding that will be stored in a file
+ * and the variable value will contain the temp file location.
+ * If this substitution occurs then an additional variable called FILE_PARAMS will contain the list of substituted variable names.
+ * @param {Object} vars 
+ * @param {Number} sizeLimit Size limit to uphold (number of bytes)
+ * @returns Object of values from the provided one, modified as described.
+ */
+function limitVariableSizes(vars, sizeLimit = CMD_SIZE_LIMIT) {
```

Comment:
This is only for cli-tools, so it could in theory go in the same file as the one used to deal with bash workers, instead of the more generic utils.